controllers/metal3.io/baremetalhost_controller_test.go

```diff
@@ -363,48 +363,76 @@ func TestHasRebootAnnotation(t *testing.T) {
 	host := newDefaultHost(t)
 	host.Annotations = make(map[string]string)
 
-	// FIXME (rdo) Here we ignore the isHardReboot boolean from
-	// hasRebootAnnotation(). We may want to expand the unit-test
-	// to validate the hard poweroff functionality as we build
-	// out such capabilities.
-	hasReboot, _ := hasRebootAnnotation(host)
-	if hasReboot {
-		t.Fail()
-	}
+	hasReboot, hardMode := hasRebootAnnotation(host)
+	assert.False(t, hasReboot)
+	assert.False(t, hardMode)
 
 	host.Annotations = make(map[string]string)
 	suffixedAnnotation := rebootAnnotationPrefix + "/foo"
 	host.Annotations[suffixedAnnotation] = ""
 
-	hasReboot, _ = hasRebootAnnotation(host)
-	if !hasReboot {
-		t.Fail()
-	}
+	hasReboot, hardMode = hasRebootAnnotation(host)
+	assert.True(t, hasReboot)
+	assert.False(t, hardMode)
 
 	delete(host.Annotations, suffixedAnnotation)
 	host.Annotations[rebootAnnotationPrefix] = ""
 
-	hasReboot, _ = hasRebootAnnotation(host)
-	if !hasReboot {
-		t.Fail()
-	}
+	hasReboot, hardMode = hasRebootAnnotation(host)
+	assert.True(t, hasReboot)
+	assert.False(t, hardMode)
 
 	host.Annotations[suffixedAnnotation] = ""
 
-	hasReboot, _ = hasRebootAnnotation(host)
-	if !hasReboot {
-		t.Fail()
-	}
+	hasReboot, hardMode = hasRebootAnnotation(host)
+	assert.True(t, hasReboot)
+	assert.False(t, hardMode)
 
 	//two suffixed annotations to simulate multiple clients
 
 	host.Annotations[suffixedAnnotation+"bar"] = ""
 
-	hasReboot, _ = hasRebootAnnotation(host)
-	if !hasReboot {
-		t.Fail()
-	}
+	hasReboot, hardMode = hasRebootAnnotation(host)
+	assert.True(t, hasReboot)
+	assert.False(t, hardMode)
+}
+
+func TestHasHardRebootAnnotation(t *testing.T) {
+	host := newDefaultHost(t)
+	host.Annotations = make(map[string]string)
+
+	hasReboot, hardMode := hasRebootAnnotation(host)
+	assert.False(t, hasReboot)
+	assert.False(t, hardMode)
+
+	host.Annotations = make(map[string]string)
+	suffixedAnnotation := rebootAnnotationPrefix + "/foo/"
+	host.Annotations[suffixedAnnotation] = "{\"mode\": \"hard\"}"
+
+	hasReboot, hardMode = hasRebootAnnotation(host)
+	assert.True(t, hasReboot)
+	assert.True(t, hardMode)
+
+	delete(host.Annotations, suffixedAnnotation)
+	host.Annotations[rebootAnnotationPrefix] = "{\"mode\": \"soft\"}"
```

Comment:
You might also want to add a test for the case where the value isn't valid json, which should also result in hardMode==false

Reply:
Done, thanks for the suggestion!